### ISSUE TITLE
`WorldQuery` triggers missing docs 

### ISSUE THREAD

**Issue by bzm3r** [bevyengine/bevy#6440]
## Bevy version

0.8, main

## What you did

I am using `#![forbid(missing_docs)]`, and trying to derive `#[derive(WorldQuery)]` on a struct . Here's the minimal not working example: https://github.com/bzm3r/world_query_missing_docs

## What went wrong

See the compiler error:

```
error: missing documentation for a struct
  --> src\main.rs:11:10
   |
11 | #[derive(WorldQuery)]
   |          ^^^^^^^^^^
   |
note: the lint level is defined here
  --> src\main.rs:1:11
   |
1  | #![forbid(missing_docs)]
   |           ^^^^^^^^^^^^
   = note: this error originates in the derive macro `WorldQuery` (in Nightly builds, run with -Z macro-backtrace for more info)
   ```

## Additional information

See discussion here: https://github.com/bevyengine/bevy/pull/4551

Either `#[automatically_derived]` no longer works as intended, or there is another struct for which it should also be tagged with `#[automatically_derived]`? Possibly:

https://github.com/bevyengine/bevy/blob/0cfdbddf2e6350b11bd9dce2d4fe89ac9c99aec9/crates/bevy_ecs/macros/src/fetch.rs#L190

**Comment by CGMossa** [comment]
See #5283

**Comment by mockersf** [comment]
adding this issue to the 0.9 milestone, it would be nice to not trigger warnings in user code

**Comment by alice-i-cecile** [comment]
Closed by #5283.
